Source/Dafny/Compiler-java.cs

```diff
@@ -3757,5 +3757,12 @@ protected override BlockTargetWriter CreateIterator(IteratorDecl iter, TargetWri
     {
       throw new NotImplementedException();
     }
+
+    // Transforms a legal file name (without extension or directcory) into 
+    // a legal class name in the target language
+    public override string TransformToClassName(string baseName) {
+      //return baseName.Replace('-', '_');
+      return System.Text.RegularExpressions.Regex.Replace(baseName, "[^_A-Za-z0-9\\$]", "_");
```

Comment:
Does this replace one occurrence or all occurrences? (I presume the latter, since the test file below has both a `-` and a `$` in the name.)